src/golang/cmd/server/handler/edit_integration.go

```diff
@@ -0,0 +1,318 @@
+	"github.com/aqueducthq/aqueduct/lib/workflow/operator/connector/auth"
+	"github.com/dropbox/godropbox/errors"
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+)
+
+var (
+	ErrNoEditPermission            = errors.New("You don't have permission to edit this integration")
+	ErrInvalidServiceType          = errors.New("Editing for this integration type is not currently supported.")
+	ErrEditDemoIntegration         = errors.New("You cannot edit demo DB credentials.")
+	ErrEditIntegrationWithDemoName = errors.New("aqueduct_demo is reserved for demo integration. Please use another name.")
+)
+
+// ConnectIntegrationHandler connects a new integration for the organization.
+type EditIntegrationHandler struct {
+	PostHandler
+
+	Database          database.Database
+	IntegrationReader integration.Reader
+	IntegrationWriter integration.Writer
+	Vault             vault.Vault
+	JobManager        job.JobManager
+	StorageConfig     *shared.StorageConfig
+}
+
+var serviceToReadOnlyFields = map[integration.Service]map[string]bool{
+	integration.Airflow:  {"host": true},
+	integration.BigQuery: {"project_id": true},
+	integration.MariaDb: {
+		"host":     true,
+		"port":     true,
+		"database": true,
+	},
+	integration.MySql: {
+		"host":     true,
+		"port":     true,
+		"database": true,
+	},
+	integration.Postgres: {
+		"host":     true,
+		"port":     true,
+		"database": true,
+	},
+	integration.Redshift: {
+		"host":     true,
+		"port":     true,
+		"database": true,
+	},
+	integration.Snowflake: {
+		"account_identifier": true,
+		"warehouse":          true,
+		"database":           true,
+	},
+}
+
+func (*EditIntegrationHandler) Headers() []string {
+	return []string{
+		routes.IntegrationNameHeader,
+		routes.IntegrationConfigHeader,
+	}
+}
+
+type EditIntegrationArgs struct {
+	*aq_context.AqContext
+	Name          string
+	IntegrationId uuid.UUID
+	UpdatedFields map[string]string
+}
+
+type EditIntegrationResponse struct{}
+
+func (*EditIntegrationHandler) Name() string {
+	return "EditIntegration"
+}
+
+// `updateConfig` updates `curConfigToUpdate` *in-place* with `newConfig` with
+// the same behavior as map updates.
+// It returns 3 values:
+//  - whether there's actually an update
+//  - http status code
+//  - error if there's any
+//
+// If trying to update a 'read only' field defined by `ServerToReadOnlyFieldsMap`,
+// `updateConfig` will return a 400 and an error.
+func updateConfig(
+	curConfigToUpdate map[string]string,
+	service integration.Service,
+	newConfig map[string]string,
+) (bool, int, error) {
+	readOnlyFields := serviceToReadOnlyFields[service]
+	updated := false
+	for k, v := range newConfig {
+		if v == "" {
+			continue // no update occurs
+		}
+
+		_, isReadonlyField := readOnlyFields[k]
+		curValue, existsInCurConfig := curConfigToUpdate[k]
+		if isReadonlyField && existsInCurConfig && curValue != v {
+			// Throw if:
+			// * field is read-only, and
+			// * field both exists in cur and new, and
```

Comment:
Is there a reason why a field would be read-only, but not exist in the current config?